xlators/mgmt/glusterd/src/glusterd-op-sm.c

```diff
@@ -2973,12 +3105,23 @@ glusterd_op_set_volume(dict_t *dict, char **errstr)
         if (global_opt) {
             cds_list_for_each_entry(voliter, &priv->volumes, vol_list)
             {
+                /* Check dependency of options before setting them */
+                ret = glusterd_dependency_chain_check(voliter, key, value,
+                                                      errstr);
+                if (ret)
+                    goto out;
+
                 value = gf_strdup(value);
                 ret = dict_set_dynstr(voliter->dict, key, value);
                 if (ret)
                     goto out;
             }
         } else {
+            /* Check dependency of options before setting them */
+            ret = glusterd_dependency_chain_check(volinfo, key, value, errstr);
+            if (ret)
+                goto out;
+
```

Comment:
The other issue is that enforcing/permissive policy should be followed also here.

Reply:
The reason for not adding the check was, that suppose a user starts glusterd in `permissive mode` (option checks are bypassed) and then they again set some options in an improper way, then they will never know about the issue or try to set the current options or the already set options in a correct way. So, if they are reminded about the issue again over here, they will be forced to fix it, instead of bypassing the options always.